TypeCobol/Compiler/Scanner/Scanner.cs

```diff
@@ -612,6 +620,93 @@ public Token GetNextToken()
                 return ScanPictureCharacterStringOrISOrSYMBOL(startIndex);
             }
 
+
+            if (currentState.InsideMultilineComments)
+            {
+                // We are inside a Multiline Comments
+                // if there is no Multiline Comments end marckup "*>>" then create a new Comment Token
+                if ((line.Length > startIndex + 2 && line.Substring(startIndex).StartsWith("*>>")) ||
+                    (line.Length > startIndex + 1 && line.Substring(startIndex).StartsWith(">>") && line[currentIndex - 1] == '*'))
+                {
+                    currentIndex = lastIndex + 1;
+                    return new Token(TokenType.CommentLine, startIndex, lastIndex, tokensLine);
+                }
+                else
+                {
+                    return ScanUntilDelimiter(startIndex, TokenType.CommentLine, "*>>");
+                }
+            }
+
+            // --- switch dedicated to formalized Comments as they are not compatible with TypeCobol grammar ---
+            if (currentState.InsideFormalizedComment)
+            {
+                switch (line[startIndex])
+                {
+                    case ' ':
+                        //SpaceSeparator=1,
+                        return ScanWhitespace(startIndex);
+                    case '-':
+                        currentIndex++;
+                        return new Token(TokenType.MinusOperator, startIndex, currentIndex - 1, tokensLine);
+                    case '@':
+                        currentIndex++;
+                        return new Token(TokenType.AtSign, startIndex, currentIndex - 1, tokensLine); 
+                    case ':':
+                        // The colon char can appears inside the formalized comments as a separator between keys and value
+                        // Consume the rest of the line and create a FloatingComment token as value for the formalized comment
+                        currentIndex = lastIndex + 1;
+                        return ScanUntilDelimiter(startIndex, TokenType.FormComsValue, "*>>>");
+                    case '*':
+                        if (line[currentIndex + 1] == '>' && line[currentIndex + 2] == '>' && line[currentIndex + 3] == '>')
+                        {
+                            // We are in the case of a Formalize Comment stop with the '*' on column other than 7
+                            // consume the * char and the three < chars
+                            currentIndex += 4;
+                            return new Token(TokenType.FormalizedCommentsStop, startIndex, startIndex + 3, tokensLine);
+                        }
+                        currentIndex ++;
+                        return new Token(TokenType.MultiplyOperator, startIndex, currentIndex - 1, tokensLine);
+                    case '>':
+                        if (line[currentIndex - 1] == '*' && line[currentIndex + 1] == '>' && line[currentIndex + 2] == '>')
```

Comment:
The previous case doesn't handle this case?

Reply:
No, as the currentIndex start at 8, the 7th column character is ignored.
So if the '*' is on column 7 then the first seen character is '>'.
In summary the case '>' catch a closing markup that begins on column 7 and the case '*' catch a closing markup that begins on column 8 or more.